.github/workflows/integration.yml

```diff
@@ -0,0 +1,105 @@
+name: integration tests
+
+on:
+  workflow_dispatch:
+  workflow_call:
+  pull_request:
+    branches:
+      - develop
+      - releases/**
+
+concurrency:
+  group: intergrations-${{github.ref}}-${{github.event.pull_request.number || github.run_number}}
+  cancel-in-progress: true
+
+jobs:
+  scan:
+    runs-on: ubuntu-latest
+    outputs:
+      packages: ${{ steps.scan.outputs.packages }}
+    steps:
+      - uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
+        with:
+          fetch-depth: 0
+      - name: "Add Integration test repo"
+        run: |
+          . share/spack/setup-env.sh
+          spack repo add ${GITHUB_WORKSPACE}/var/spack/test_repos/spack_repo/integration_test
+      - name: "List repo packages"
+        id: scan
+        run: |
+          . share/spack/setup-env.sh
+          packages_list="$(spack list --repo integration_test | python3 -c "import sys; print([f'"{x}"' for x in sys.stdin.read().split('\n') if x])")"
+          echo "packages=${packages_list}" >> "$GITHUB_OUTPUT"
```

Comment:
I recommend adding a TODO: there might be packages that we're only interested in as dependencies of other packages. The way concretization works, this could end up installing `O(n^2)` packages for a repo of size `N`: you get effects like this for example if for `x->y`, `x` has `depends_on(y~default_on)`; `install x` and `install y` will generate 3 specs.

Overall, I think that adding new packages to the integration test repo will be rare enough that it could be fine to list them out explicitly.